src/main/java/com/google/sps/DataGraph.java

```diff
@@ -140,6 +172,8 @@ public boolean mutateGraph(Mutation mut) {
     GraphNode startNode = graphNodesMap.get(startName);
     GraphNode endNode = graphNodesMap.get(endName);
 
+    Set<GraphNode> successors;
```

Comment:
With the change I suggested below to place the `graph.successors` within the `if` header, you shouldn't need to declare the `successors` variable outside of the switch statement anymore. 

I assume you did it because otherwise it would be a compile error that variable `successors` is already defined. This is because Java case statements have a weird notion of scope - by default the `case` statements share a scope, which is a bit unintuitive. This is why I like to manually introduce a new scope for my case statements by adding extra `{}` so that you can do something like:
```
switch (mut.getType()) {
    case ADD_NODE: {
         int x = 1;
         break
    }
    case ADD_EDGE: {
         int x = 2;
         break
    }
}
```
The two `x`s above are now in different scopes :)

Reply:
Oh that's a really good idea! Yeah I was so confused by that compile error, it's very unintuitive.

Reply:
Oh whoa I did not know this!!